glue_jupyter/bqplot/image/layer_artist.py

```diff
@@ -1,182 +1,82 @@
-import numpy as np
-import bqplot
-from ipyastroimage.astroimage import AstroImage
-import ipywidgets as widgets
-import ipywidgets.widgets.trait_types as tt
-import matplotlib.cm
+from glue.viewers.image.layer_artist import BaseImageLayerArtist, ImageLayerArtist, ImageSubsetArray
+from glue.viewers.image.state import ImageSubsetLayerState
+from glue.core.fixed_resolution_buffer import ARRAY_CACHE, PIXEL_CACHE
 
-from glue.core.exceptions import IncompatibleAttribute
-from glue.core.data import Subset
-from glue.viewers.image.state import ImageLayerState
-from glue.viewers.image.layer_artist import ImageLayerArtist
-from glue_jupyter.compat import LayerArtist
+from .frb_mark import FRBImage
 
-from ...link import link, on_change
-from ...widgets import LinkedDropdown
 
-# FIXME: monkey patch ipywidget to accept anything
-tt.Color.validate = lambda self, obj, value: value
+class BqplotImageLayerArtist(ImageLayerArtist):
 
+    def enable(self):
+        if self.enabled:
+            return
 
-colormaps = [('Viridis', 'viridis'), ('Jet', 'jet'), ('Grey', ['black', 'grey']), ('RdYlGn', 'RdYlGn')]
+    def redraw(self):
+        pass
+
+
+class BqplotImageSubsetLayerArtist(BaseImageLayerArtist):
+
+    _layer_state_cls = ImageSubsetLayerState
+
+    def __init__(self, axes, viewer_state, layer_state=None, layer=None):
+
+        super(BqplotImageSubsetLayerArtist, self).__init__(axes, viewer_state,
+                                                           layer_state=layer_state, layer=layer)
+
+        self.subset_array = ImageSubsetArray(self._viewer_state, self)
+
+        self.image_artist = FRBImage(axes, self.subset_array)
```

Comment:
A bit confused here, the first argument is axes, which is passed along here, but in FRBImage it is viewer, shouldn't this be viewer? (copy paste mistake from matplotlib code maybe?)

Reply:
Yes it was from copy/pasting - will fix